akka-distributed-data/src/main/scala/akka/cluster/ddata/ORMultiMap.scala

```diff
@@ -69,9 +69,12 @@ final class ORMultiMap[A, B] private[akka] (
 
   override def merge(that: T): T =
     if (withValueDeltas == that.withValueDeltas) {
-      if (withValueDeltas)
-        new ORMultiMap(underlying.mergeRetainingDeletedValues(that.underlying), withValueDeltas)
-      else
+      if (withValueDeltas) {
+        val newUnderlying = underlying.mergeRetainingDeletedValues(that.underlying)
+        // Garbage collect the tombstones we no longer need, i.e. those that have Set() as a value.
+        val newValues = newUnderlying.values.filterNot(elem ⇒ (!newUnderlying.keys.contains(elem._1)) && elem._2.isEmpty)
```

Comment:
`filterNot { case (key, value) =>`

Reply:
Thanks. Tried to figure it out, but forgot, that in Scala you need to use curly braces (`{`) to enable this. Why does this language need to have so irregular grammar...

Reply:
Martin would disagree :D But yeah, the dichotomy between partial and functions is no fun and will hopefully be addressed at some point sooner rather than later

Reply:
I am probably too harsh here, but my first 'serious' programming language (after BASIC and 6502 Assembly) was Pascal, and I am missing the consistency and elegance ever since :)